tests/components/min_max/test_sensor.py

```diff
@@ -14,329 +13,334 @@
     TEMP_CELSIUS,
     TEMP_FAHRENHEIT,
 )
-from homeassistant.setup import async_setup_component, setup_component
+from homeassistant.setup import async_setup_component
 
 from tests.async_mock import patch
-from tests.common import get_test_home_assistant
-
-
-class TestMinMaxSensor(unittest.TestCase):
-    """Test the min/max sensor."""
-
-    def setup_method(self, method):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.values = [17, 20, 15.3]
-        self.count = len(self.values)
-        self.min = min(self.values)
-        self.max = max(self.values)
-        self.mean = round(sum(self.values) / self.count, 2)
-        self.mean_1_digit = round(sum(self.values) / self.count, 1)
-        self.mean_4_digits = round(sum(self.values) / self.count, 4)
-        self.median = round(statistics.median(self.values), 2)
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_min_sensor(self):
-        """Test the min sensor."""
-        config = {
-            "sensor": {
-                "platform": "min_max",
-                "name": "test_min",
-                "type": "min",
-                "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
-            }
+
+VALUES = [17, 20, 15.3]
+COUNT = len(VALUES)
+MIN_VALUE = min(VALUES)
+MAX_VALUE = max(VALUES)
+MEAN = round(sum(VALUES) / COUNT, 2)
+MEAN_1_DIGIT = round(sum(VALUES) / COUNT, 1)
+MEAN_4_DIGITS = round(sum(VALUES) / COUNT, 4)
+MEDIAN = round(statistics.median(VALUES), 2)
+
+
+async def test_min_sensor(hass):
+    """Test the min sensor."""
+    config = {
+        "sensor": {
+            "platform": "min_max",
+            "name": "test_min",
+            "type": "min",
+            "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
         }
+    }
+
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
 
-        assert setup_component(self.hass, "sensor", config)
+    entity_ids = config["sensor"]["entity_ids"]
 
-        entity_ids = config["sensor"]["entity_ids"]
+    for entity_id, value in dict(zip(entity_ids, VALUES)).items():
+        hass.states.async_set(entity_id, value)
+        await hass.async_block_till_done()
 
-        for entity_id, value in dict(zip(entity_ids, self.values)).items():
-            self.hass.states.set(entity_id, value)
-            self.hass.block_till_done()
+    state = hass.states.get("sensor.test_min")
 
-        state = self.hass.states.get("sensor.test_min")
+    assert str(float(MIN_VALUE)) == state.state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)